src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.LocalDate;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(storage.load());
+            tasks.list();
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                if (fullCommand.equals("bye")) {
+                    ArrayList<String> data = tasks.bye();
+                    storage.save(data);
+                    ui.showBye();
+                    break;
+                } else {
+                    boolean exit = Parser.parse(fullCommand, tasks, ui, storage);
+                    isExit = exit;
+                }
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke(System.getProperty("user.dir") + "/data/duke.txt").run();
+    }
+
+}
+
```

Comment:
Overall code is very clear and should have no problem finishing up the ip in due time